source/gloperate/source/stages/base/ShaderStage.cpp

```diff
@@ -30,6 +30,7 @@ ShaderStage::~ShaderStage()
 void ShaderStage::onProcess(AbstractGLContext *)
 {
     m_shader = std::unique_ptr<globjects::Shader>(environment()->resourceManager()->load<globjects::Shader>((*filePath).path()));
+    assert(m_shader);
```

Comment:
I'm not sure about the assert. We cannot make sure that m_shader is never nullptr, because it depends on user input. If a wrong filename is chosen, m_shader will be nullptr, so the output of the stage must also be nullptr. It's up to the other stages to deal with that (e.g., a program stage). However I don't think I want to hit an assertion if I as a am running the shader editor in debug mode and happen to enter a wrong filename for a shader ...